### PR TITLE
Parse autoinstall journal events

### LINKED ISSUE #1454: Autoinstall: copying files
The GUI is normally able to detect what Subiquity is doing, and presents different status messages during different phases of the installation process.

image
image

During automated installations, however, the message seems to be always "Copying files" which is a fallback used when the GUI doesn't know what Subiquity is doing:

image

### PR DESCRIPTION
**Pull request by d-loose**
Fix #1454

### CODE REVIEW

File: packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart [hunk 1]
```diff
@@ -21,6 +22,17 @@ class InstallationEvent {
       description: description ?? this.description,
     );
   }
+
+  String localize(AppLocalizations lang) {
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart:26]: i'd prefer to keep translations out of the model. could we use an enum instead?

File: packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_page.dart [hunk 2]
```diff
@@ -145,7 +133,9 @@ class _InstallationSlidesPageState extends State<InstallationSlidesPage> {
                     Text(
                       model.hasError
                           ? lang.installationFailed
-                          : _formatEvent(model.event),
+                          : (model.event ??
+                                  const InstallationEvent('copying files'))
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_page.dart:137]: perhaps we could return this as a fallback from the model?

File: packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart [hunk 3]
```diff
@@ -70,12 +82,31 @@ class InstallationSlidesModel extends SafeChangeNotifier {
...
   //   downloading and installing security updates
   // ```
+  // In autoinstall mode syslog events start with "subiquity" and unindented
+  // entries contain no useful information.
+  // ```
+  // subiquity/Drivers/_list_drivers
+  //   subiquity/Install/install/unmount_target: umounting /target dir
+  //   subiquity/Install/install/curtin_install: installing system
+  //     subiquity/Install/install/curtin_install/run: executing curtin install initial step
+  //     subiquity/Install/install/curtin_install/run: executing curtin install partitioning step
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart:92]: could we add a test with these lines?

File: packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart [hunk 4]
```diff
@@ -70,12 +82,31 @@ class InstallationSlidesModel extends SafeChangeNotifier {
...
+  // ```
   void _processEvent(String syslog) {
     final trimmed = syslog.trimLeft();
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart:95]: a crazy idea; what if we would do something like
```dart
syslog.removeOne(RegExp(r'  subiquity/[\w/]+:'))
```
as the first step? would the rest of the function work as before?

Comment (jpnurmi) [reply packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart:95]: i meant `replaceFirst(rx, '')` :)

Comment (d-loose) [reply packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart:107]: good idea - works

**Review by jpnurmi — APPROVED** [review]
Nice! LGTM :+1:

**Pull request merged**